##<font color='Green'> **Loading** **Dataset** **Autolib** **and** **Importing** **libraries**.


In [ ]:
import pandas as pd

df = pd.read_csv('/content/Autolib_dataset.csv')
df

Address  Cars  ...  hour  minute
0                    2 Avenue de Suffren     0  ...    11      43
1              145 Rue Raymond Losserand     6  ...     7      24
2       2 Avenue John Fitzgerald Kennedy     3  ...    20      14
3                 51 Rue Eugène Oudiné     3  ...     4      37
4     6 avenue de la Porte de Champerret     3  ...    17      23
...                                  ...   ...  ...   ...     ...
4995                 8 avenue Ménelotte     2  ...    11      26
4996                   37 rue de Dantzig     4  ...    16      56
4997                      142 rue du Bac     1  ...     7       1
4998         2 avenue du Val de Fontenay     2  ...    17      27
4999                 5 Rue Henry Brisson     4  ...    10       2

[5000 rows x 25 columns]

In [ ]:
df.shape

(5000, 25)

# <font color='Green'> **Data** **Cleaning**

In [ ]:
df.drop(['Geo point','Scheduled at','Displayed comment'], axis = 1,inplace = True)

In [ ]:
df.head()

Address  Cars  Bluecar counter  ...  day  hour  minute
0                 2 Avenue de Suffren     0                0  ...    8    11      43
1           145 Rue Raymond Losserand     6                6  ...    6     7      24
2    2 Avenue John Fitzgerald Kennedy     3                3  ...    3    20      14
3              51 Rue Eugène Oudiné     3                3  ...    4     4      37
4  6 avenue de la Porte de Champerret     3                3  ...    8    17      23

[5 rows x 22 columns]

In [ ]:
df.drop_duplicates().any()

Address                True
Cars                   True
Bluecar counter        True
Utilib counter         True
Utilib 1.4 counter     True
Charge Slots           True
Charging Status        True
City                   True
ID                     True
Kind                   True
Postal code            True
Public name            True
Rental status          True
Slots                  True
Station type           True
Status                 True
Subscription status    True
year                   True
month                  True
day                    True
hour                   True
minute                 True
dtype: bool

In [ ]:
df.isnull().sum()

Address                0
Cars                   0
Bluecar counter        0
Utilib counter         0
Utilib 1.4 counter     0
Charge Slots           0
Charging Status        0
City                   0
ID                     0
Kind                   0
Postal code            0
Public name            0
Rental status          0
Slots                  0
Station type           0
Status                 0
Subscription status    0
year                   0
month                  0
day                    0
hour                   0
minute                 0
dtype: int64

In [ ]:
df['Date_time'] = pd.to_datetime(df[['year','month','day','hour','minute']],format='%Y:%M:%D:%H:%M')
df

Address  Cars  ...  minute           Date_time
0                    2 Avenue de Suffren     0  ...      43 2018-04-08 11:43:00
1              145 Rue Raymond Losserand     6  ...      24 2018-04-06 07:24:00
2       2 Avenue John Fitzgerald Kennedy     3  ...      14 2018-04-03 20:14:00
3                 51 Rue Eugène Oudiné     3  ...      37 2018-04-04 04:37:00
4     6 avenue de la Porte de Champerret     3  ...      23 2018-04-08 17:23:00
...                                  ...   ...  ...     ...                 ...
4995                 8 avenue Ménelotte     2  ...      26 2018-04-06 11:26:00
4996                   37 rue de Dantzig     4  ...      56 2018-04-04 16:56:00
4997                      142 rue du Bac     1  ...       1 2018-04-01 07:01:00
4998         2 avenue du Val de Fontenay     2  ...      27 2018-04-04 17:27:00
4999                 5 Rue Henry Brisson     4  ...       2 2018-04-08 10:02:00

[5000 rows x 23 columns]

In [ ]:
df.drop_duplicates().any()

Address                True
Cars                   True
Bluecar counter        True
Utilib counter         True
Utilib 1.4 counter     True
Charge Slots           True
Charging Status        True
City                   True
ID                     True
Kind                   True
Postal code            True
Public name            True
Rental status          True
Slots                  True
Station type           True
Status                 True
Subscription status    True
year                   True
month                  True
day                    True
hour                   True
minute                 True
Date_time              True
dtype: bool

In [ ]:
df.isnull().sum()

Address                0
Cars                   0
Bluecar counter        0
Utilib counter         0
Utilib 1.4 counter     0
Charge Slots           0
Charging Status        0
City                   0
ID                     0
Kind                   0
Postal code            0
Public name            0
Rental status          0
Slots                  0
Station type           0
Status                 0
Subscription status    0
year                   0
month                  0
day                    0
hour                   0
minute                 0
Date_time              0
dtype: int64

In [ ]:
df.shape

(5000, 23)

# <font color='Green'> **Data** **Analysis**

In [ ]:
# Identify the most popular hour of the day for picking up a shared electric car 
# (Bluecar) in the city of Paris over the month of April 2018.

x = df[(df['Charging Status']== 'operational')& (df['City']=='Paris')]['Bluecar counter'].value_counts().max()
df.loc[x]['Date_time']

Timestamp('2018-04-08 09:42:00')

<font color='Green'> In the above analysis the most popular hour for renting cars is 8:15 where the cars are maximum at the station and that means the station is supposed to be operational also city should be Paris.

In [ ]:
# What is the most popular hour for returning cars?
x=df[(df['Charging Status']== 'operational') & (df['Rental status']== 'operational')]['Bluecar counter'].value_counts().min()
df.loc[x]['Date_time']

Timestamp('2018-04-06 07:02:00')

<font color='Green'>The above analysis shows that the time for returning cars is 07:02   the time when cars for renting are minimun in the station and also point to not is that a station should be operatinal.

In [ ]:
# What station is the most popular?
# Overall?
x = df[(df['Charging Status']== 'operational')&(df['Rental status']=='operational')]['Bluecar counter'].value_counts().max()
df.loc[x]['Public name']

'Paris/Olier/15'

<font color='Green'>The most popular station should have the maximum cars and should be operational that is "Clamart/Ferrari/1"

In [ ]:
# At the most popular picking hour?
x =df[(df['Charging Status']== 'operational')&(df['Rental status']=='operational')]['Date_time'].value_counts().max()
df.loc[x]['Public name']

'Paris/Eugène Oudiné/51'

In [ ]:
#What postal code is the most popular for picking up Blue cars? overall?
x =df[(df['Charging Status']== 'operational')&(df['Rental status']=='operational')]['Bluecar counter'].value_counts().max()
df.loc[x]['Postal code']


75015

In [ ]:
#Does the most popular station belong to that postal code?
df.loc[x]

Address                       15 rue Olier
Cars                                     4
Bluecar counter                          4
Utilib counter                           0
Utilib 1.4 counter                       0
Charge Slots                             0
Charging Status                operational
City                                 Paris
ID                          paris-olier-15
Kind                               STATION
Postal code                          75015
Public name                 Paris/Olier/15
Rental status                  operational
Slots                                    1
Station type                       station
Status                                  ok
Subscription status            nonexistent
year                                  2018
month                                    4
day                                      6
hour                                    22
minute                                  29
Date_time              2018-04-06 22:29:00
Name: 560, 

<font color='Green'>from above analysis the most popular postal code is 75015, and the postal name is"Paris/Eugène Oudiné/51" NO it belongs to the Postal code.

In [ ]:
#What postal code is the most popular for picking up Blue cars?
#At the most popular picking hour?
x =df.loc[df['Rental status']=='operational'][df['Bluecar counter']>0]['Date_time'].value_counts().max()
df.loc[x]['Postal code']

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  This is separate from the ipykernel package so we can avoid doing imports until


75017

### USING UTILIB COUNTER

In [ ]:
# Identify the most popular hour of the day for picking up a shared electric car 
# (Utilib) in the city of Paris over the month of April 2018.
x = df[(df['Charging Status']== 'operational')& (df['City']=='Paris')]['Utilib counter'].value_counts().max()
df.loc[x]['Date_time']


Timestamp('2018-04-05 19:47:00')

In [ ]:
# What is the most popular hour for returning cars?(Utilib)
x=df[(df['Charging Status']== 'operational') & (df['Rental status']== 'operational')]['Utilib counter'].value_counts().min()
df.loc[x]['Date_time']

Timestamp('2018-04-06 07:02:00')

In [ ]:
# What station is the most popular?
# Overall? (for Utilib)
x = df[(df['Charging Status']== 'operational')&(df['Rental status']=='operational')]['Utilib counter'].value_counts().max()
df.loc[x]['Public name']

'Créteil/Juliette Savar/102'

In [ ]:
#What postal code is the most popular for picking up Utilib? overall?
x =df[(df['Charging Status']== 'operational')&(df['Rental status']=='operational')]['Utilib counter'].value_counts().max()
df.loc[x]['Postal code']

94000

### USING UTILIB 1.4 COUNTER

In [ ]:
# Identify the most popular hour of the day for picking up a shared electric car 
# (Utilib 1.4) in the city of Paris over the month of April 2018.
x = df[(df['Charging Status']== 'operational')& (df['City']=='Paris')]['Utilib 1.4 counter'].value_counts().max()
df.loc[x]['Date_time']

Timestamp('2018-04-04 04:09:00')

In [ ]:
# What is the most popular hour for returning cars?(Utilib 1.4)
x=df[(df['Charging Status']== 'operational') & (df['Rental status']== 'operational')]['Utilib 1.4 counter'].value_counts().min()
df.loc[x]['Date_time']

Timestamp('2018-04-06 07:24:00')

In [ ]:
# What station is the most popular?
# Overall? (for Utilib 1.4)
x = df[(df['Charging Status']== 'operational')&(df['Rental status']=='operational')]['Utilib 1.4 counter'].value_counts().max()
df.loc[x]['Public name']

'Paris/Choisy/188'

In [ ]:
#What postal code is the most popular for picking up Utilib 1.4 ? overall?
x =df[(df['Charging Status']== 'operational')&(df['Rental status']=='operational')]['Utilib 1.4 counter'].value_counts().max()
df.loc[x]['Postal code']

75013